src/vscrollpanel.cpp

```diff
@@ -81,11 +81,24 @@ void VScrollPanel::draw(NVGcontext *ctx) {
     mChildPreferredHeight = child->preferredSize(ctx).y();
     float scrollh = height() *
         std::min(1.0f, height() / (float) mChildPreferredHeight);
+        
+    float offset = -mScroll*(mChildPreferredHeight - mSize.y());
 
     nvgSave(ctx);
     nvgTranslate(ctx, mPos.x(), mPos.y());
     nvgIntersectScissor(ctx, 0, 0, mSize.x(), mSize.y());
-    nvgTranslate(ctx, 0, -mScroll*(mChildPreferredHeight - mSize.y()));
+    nvgTranslate(ctx, 0, offset);
+
+
+    Widget *parentWindow = (Widget*)window();
```

Comment:
Why not directly try to cast the parent window into a ``Popup``? I don't understand the need for a loop here.